CTM360 CyberBlindspot New Pack

edx-sayed-salem opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
New pack for CTM360's CyberBlindspot. Using a two-way mirroring implementation to sync with the remote platform. It contains:
- 1 integration with 4 commands.
- 1 incident type.
- 1 incident field.
- ~~1 layout.~~
- 1 dashboard.
- 1 default playbook.
- ~~1 classifier.~~
- 1 mapper.

## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=32381) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=32381) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=32381) it.</sub>

yucohen reviewed (changes requested):
Hi @edx-sayed-salem THanks for your contribution! Great work!
Please see my comments.
Also, lets set up a demo next week

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/CyberBlindspot.py:
```diff
@@ -0,0 +1,807 @@
+            log(INFO, 'Invalid "API Key" Value')
+            raise DemistoException('Invalid "API Key" Value')
+        incidents = client.test_configuration(args)
+        if max_fetch and len(incidents) > max_fetch:
+            log(INFO, 'Result count does not adhere to "Max Fetch" value')
+            raise DemistoException('Result count does not adhere to "Max Fetch" value')
```
If it's less or equal its fine since the condition only triggers for more than max_fetch. Do I need to change anything?

yucohen replied:
If there are more incidents than the max_fetch, then we need to fetch the first batch and in the next fetch to get the rest of the incidents

edx-sayed-salem replied:
Done in [37dc2dc](https://github.com/demisto/content/pull/32381/commits/37dc2dc6fc0a2bbfc263df24caae6f77bb4c1b8c)

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/CyberBlindspot.py:
```diff
@@ -0,0 +1,807 @@
+    """
+    log(DEBUG, "at main func")
+
+    api_key = demisto.params().get('api_key', {}).get('password')
```
Done in [da8f96d](https://github.com/demisto/content/pull/32381/commits/da8f96ddf9afcfa5529dc92d64d6b80f34c05a70)

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/CyberBlindspot.py:
```diff
@@ -0,0 +1,807 @@
+        # if input_format_string:
+        #     output = arg_to_datetime(parse_date_string(time_string, input_format_string).isoformat(), **parser_args)
+        # else:
+        #     output = arg_to_datetime(time_string, **parser_args)
```
Done in [f076f61](https://github.com/CTM360-Integrations/content/commit/f076f613e895d368c4ea8e750e7058666ecb7e08)

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/CyberBlindspot.py:
```diff
@@ -0,0 +1,807 @@
+MAX_RETRIES = 5
+MIRROR_LIMIT = 1000
+ABSOLUTE_MAX_FETCH = 200
+MAX_FETCH = demisto.params().get('max_fetch', 0)
```
The below suggestion was applied

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/CyberBlindspot.py:
```diff
@@ -0,0 +1,807 @@
+            params = {
+                'date_field': '@timestamp',
+                'order': 'asc',
+                'max_hits': MAX_FETCH,
```
Done in [da8f96d](https://github.com/demisto/content/pull/32381/commits/da8f96ddf9afcfa5529dc92d64d6b80f34c05a70)

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/test_data/empty_list.json:
```diff
@@ -0,0 +1 @@
+[]
```
Done in [3b5a4d4](https://github.com/demisto/content/pull/32381/commits/3b5a4d4fa028e9a4be97ce52af63c6f2e5b24e91) and [c2715ca](https://github.com/demisto/content/pull/32381/commits/c2715ca1dff68fbbcdc2457a7b9593e6ca6e33a6)

edx-sayed-salem commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/test_data/empty_object.json:
```diff
@@ -0,0 +1 @@
+{}
```
Done in [3b5a4d4](https://github.com/demisto/content/pull/32381/commits/3b5a4d4fa028e9a4be97ce52af63c6f2e5b24e91) and [c2715ca](https://github.com/demisto/content/pull/32381/commits/c2715ca1dff68fbbcdc2457a7b9593e6ca6e33a6)

melamedbn commented:
Hi @edx-sayed-salem,

I've reviewed the CTM360 content, and here are some points for consideration:

### Incoming Mapper
- **CustomFields.cbs_coa Field**: Is this field created upon fetching? It would be helpful to see the mapping in a demo.

### Classifier
- **Classification Verification**: It appears that no actual classification has been made. Could you please double-check to ensure everything is set up correctly?

### CyberBlindspot Incident Layout
- **Naming**: Consider removing "Layout" from the name for clarity and conciseness.
- **CTM360 Specific Elements**: I didn't notice any CTM360-specific objects in the layout. Could you clarify the rationale behind adding this to the marketplace?

### CyberBlindspot Default
- **Playbook Naming**: Let's think about a more descriptive name than 'default' to better reflect the playbook's purpose.
- **Playbook Description**: Could you enhance the description to provide more clarity on the playbook's functionality?
- **Quiet Mode Operation**: Are you sure about setting the playbook to run in quiet mode? It's important to consider the visibility of operations.
[Quiet mode](https://xsoar.pan.dev/docs/playbooks/playbook-settings)
- **First Task - Extract Indicators**: Since the incident type is set to automatically extract all indicators, the initial 'extract indicators' task may not be necessary, which could improve performance.
- **Playbook Outputs**: The playbook currently doesn't have outputs. Is this intentional? Adding outputs might be beneficial for tracking and further processing.

Looking forward to the upcoming demo and your thoughts on these suggestions.

Best regards,

Ben

edx-sayed-salem commented:
Hi @melamedbn,

> ### Incoming Mapper
> * **CustomFields.cbs_coa Field**: Is this field created upon fetching? It would be helpful to see the mapping in a demo.

yeah, I have a function to map the incidents upon fetch. `map_and_create_incident()`

> ### Classifier
> * **Classification Verification**: It appears that no actual classification has been made. Could you please double-check to ensure everything is set up correctly?

Since we are just gonna call every incident as 'CyberBlinspot Incident' I only needed the classifier to bind with this incident type any incoming incident from my pack. There was talk in the beginning (Design Doc) about the different types of incidents and we concluded that using 1 type was fine.

> ### CyberBlindspot Incident Layout
> * **Naming**: Consider removing "Layout" from the name for clarity and conciseness.

Sure, no problem.

> * **CTM360 Specific Elements**: I didn't notice any CTM360-specific objects in the layout. Could you clarify the rationale behind adding this to the marketplace?

Yea, that actually slipped my mind. So there was a new tab I added with custom action buttons with display filters and whatnot, however, as I worked on the playbook I saw that having the playbook do prompts is way easier on the user and so I hid them with a condition around the `runStatus` so that if the playbook is there, no need to show it. I did forget it was there as time went on.

> ### CyberBlindspot Default
> * **Playbook Naming**: Let's think about a more descriptive name than 'default' to better reflect the playbook's purpose.
> * **Playbook Description**: Could you enhance the description to provide more clarity on the playbook's functionality?

Sure, we'll try.

> * **Quiet Mode Operation**: Are you sure about setting the playbook to run in quiet mode? It's important to consider the visibility of operations.

I don't know for sure.

> * **First Task - Extract Indicators**: Since the incident type is set to automatically extract all indicators, the initial 'extract indicators' task may not be necessary, which could improve performance.

I think I added this, because the incidents I had already fetched at some point didn't have that set and I thought adding it to the playbook was good. I'll see to it.

> * **Playbook Outputs**: The playbook currently doesn't have outputs. Is this intentional? Adding outputs might be beneficial for tracking and further processing.

I had one task use a command that had an output. I didn't feel I needed more.

> Looking forward to the upcoming demo and your thoughts on these suggestions.

Yes, we can discuss more about those.

yucohen commented:
@edx-sayed-salem Great demo today!
a minor change from my side

yucohen commented on Packs/CTM360-CyberBlindspot/Integrations/CyberBlindspot/CyberBlindspot.yml:
```diff
@@ -0,0 +1,147 @@
+      name: order
+      required: false
+    outputs:
+    - contextPath: CyberBlindspot.IncidentList
+      description: List of all CBS incidents.
```
```suggestion
    - contextPath: CyberBlindspot.IncidentList.id
      description: The incident ID.
```

yucohen replied:
this is the way to add the context outputs. Just add more data according to the outputs from the api response.

edx-sayed-salem replied:
appreciate it, will do.

edx-sayed-salem replied:
Done in [329e3d4](https://github.com/demisto/content/pull/32381/commits/329e3d49898043c51a61083572cdc21a1f70f09f)

yucohen commented on Packs/RTIR/Integrations/RTIR/README.md:
```diff
@@ -120,6 +120,14 @@
 <td style="width: 589px;">Final priority of ticket</td>
 </tr>
 <tr>
+<td style="width: 132px;">member-of</td>
+<td style="width: 589px;">Ticket MembersOF links</td>
+</tr>
+<tr>
+<td style="width: 132px;">members</td>
+<td style="width: 589px;">Ticket Members links</td>
+</tr>
+<tr>
```
please revert the file

edx-sayed-salem replied:
Reverted all files in that commit in [a445f68](https://github.com/demisto/content/pull/32381/commits/a445f68238d346069202f69ee724b03caebaca16)

yucohen commented on Packs/RTIR/Integrations/RTIR/RTIR.py:
```diff
@@ -150,6 +150,28 @@ def create_ticket_request(encoded):
     return ticket_id
 
 
+def edit_links(ticket_id, member_of, members, depends_on, depended_on_by, refers_to, referred_to_by):
+
+    content = ""
+    if member_of:
+        content = f"MemberOf: {member_of}\n"
+    if members:
+        content += f"Members: {members}\n"
+    if depends_on:
+        content += f"DependsOn: {depends_on}\n"
+    if depended_on_by:
+        content += f"DependedOnBy: {depended_on_by}\n"
+    if refers_to:
+        content += f"RefersTo: {refers_to}\n"
+    if referred_to_by:
+        content += f"ReferredToBy: {referred_to_by}\n"
+
+    data = f"content={urllib.parse.quote_plus(content)}"
+
+    suffix_url = f'ticket/{ticket_id}/links'
+    return http_request('POST', suffix_url, data=data)
+
```
please revert

yucohen commented on Packs/RTIR/Integrations/RTIR/RTIR.yml:
```diff
@@ -90,6 +90,24 @@ script:
       name: initial-priority
     - description: Ticket final priority, given as number in the range 0-100.
       name: final-priority
+    - description: Ticket MemberOF ID.
+      isArray: true
+      name: member-of
+    - description: Ticket Members ID.
+      isArray: true
+      name: members
+    - description: Ticket DependedOn ID.
+      isArray: true
+      name: depends-on
+    - description: Ticket DependedOnBy ID.
+      isArray: true
+      name: depended-on-by
+    - description: Ticket RefersTo ID.
+      isArray: true
+      name: refers-to
+    - description: Ticket ReferredToBy ID.
+      isArray: true
+      name: referred-to-by
```
please revert

yucohen commented on Packs/RTIR/Integrations/RTIR/RTIR_test.py:
```diff
@@ -284,3 +284,24 @@ def test_add_comment_fail(mocker):
     with pytest.raises(DemistoException) as e:
         add_comment()
     assert str(e.value) == 'Failed to add comment'
+
+
+def test_edit_ticket(mocker):
+    """
+    Test failure in edit a ticket if no fields was given.
+
+    Given:
+    - ticket id
+
+    When:
+    - editing a ticket
+
+    Then:
+    - Ensure the command fails with an error message.
+    """
+    from RTIR import edit_ticket
+    mocker.patch.object(demisto, 'args', return_value={'ticket-id': '1234'})
+    try:
+        edit_ticket()
+    except Exception as ex:
+        assert ex.message == 'No arguments were given to edit the ticket.'
```
please revert

yucohen commented on Packs/RTIR/ReleaseNotes/1_0_20.md:
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### RTIR
+
+- added the *members*, *member-of*, *depends-on*, *depended-on-by*, *refers-to* and *referred-to-by* arguments to **rtir-create-ticket** and  **rtir-edit-ticket** commands to allow setting of the links fields.
+- Updated the Docker image to: *demisto/python3:3.10.13.85667*.
```
please revert

yucohen commented on Packs/RTIR/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "description": "Request Tracker for Incident Response is a ticketing system which provides pre-configured queues and workflows designed for incident response teams.",
     "support": "xsoar",
-    "currentVersion": "1.0.19",
+    "currentVersion": "1.0.20",
```
please revert
